cluster-autoscaler/cloudprovider/openstack/README.md

```diff
@@ -0,0 +1,46 @@
+# Cluster Autoscaler on OpenStack
+The cluster autoscaler on OpenStack scales worker nodes within any
+specified nodegroup. It will run as a `Deployment` in your cluster.
+This README will go over some of the necessary steps required to get
+the cluster autoscaler up and running.
+
+## Permissions and credentials
+
+The autoscaler needs a `ServiceAccount` with permissions for Kubernetes and
+requires credentials for interacting with OpenStack.
+
+An example `ServiceAccount` is given in [examples/os-autoscaler-svcaccount.yaml](examples/os-autoscaler-svcaccount.yaml).
+
+The credentials for authenticating with OpenStack are stored in a secret and
+mounted as a file inside the container. [examples/os-autoscaler-secret](examples/os-autoscaler-secret.yaml)
+can be modified with the contents of your cloud-config. This file can be obtained from your master node,
+in `/etc/kubernetes` (may be named `kube_openstack_config` instead of `cloud-config`).
+
+## Autoscaler deployment
+
+The deployment in `examples/os-autoscaler-deployment.yaml` can be used,
+but the arguments passed to the autoscaler will need to be changed
+to match your cluster.
+
+| Argument         | Usage                                                                                                                                      |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| --cluster-name   | The name of your Kubernetes cluster. If there are multiple clusters sharing the same name then the cluster IDs should be used instead.     |
+| --cloud-provider | Can be omitted if the autoscaler is built with `BUILD_TAGS=openstack`.                                                                     |
+| --nodes          | Of the form `min:max:NodeGroupName`. Node groups are not yet implemented in Magnum so only a single node group is currently supported.     |
```

Comment:
This the plan to drop this arg once node group support is added in magnum? If magnum provides the node group list and the min/max for each we do not need this?

Reply:
When nodegroups are being used, yes. Then either all the nodegroup information will come from magnum, or it will use the node-group-auto-discovery flag ([example](https://github.com/kubernetes/autoscaler/blob/d538ac0098b646186477fd4494c223692e089df0/cluster-autoscaler/cloudprovider/aws/examples/cluster-autoscaler-autodiscover.yaml#L140)).